### ISSUE TITLE
Request for the GSM8K-specific fuser checkpoint used for Table 15

### ISSUE THREAD

**Issue by zhangyun04** [thu-nics/C2C#15]
> Hi @giedlinlofty118-blip, thanks for the detailed follow-up. I rechecked the experiment settings for the GSM8K Table 15 run, and I need to correct one important point in my previous reply.
> 
> The main source of confusion is the fuser checkpoint. The **62.55** GSM8K C2C result in Table 15 was obtained with a **GSM8K-specific C2C fuser** trained for the `Qwen3-0.6B` receiver and `Qwen3-4B` sharer. It was not obtained with the public generic `qwen3_0.6b+qwen3_4b_Fuser` checkpoint.
> 
> ## Exact Setup for the 62.55 C2C Row
> 
> - Receiver/base model: `Qwen/Qwen3-0.6B`
> - Sharer model: `Qwen/Qwen3-4B`
> - Fuser: a GSM8K-specific C2C fuser trained from Qwen3-4B generated GSM8K training data
> - Evaluation dataset: full `openai/gsm8k` test split
> 
> So the key correction is:
> 
> **Table 15 used a task-specific GSM8K fuser, not the public general-purpose Qwen3-0.6B + Qwen3-4B fuser.**
> 
> Below are the settings used for that run.
> 
> ## 1. Teacher Data Generation for the GSM8K Fuser
> 
> The GSM8K-specific fuser was trained on GSM8K train examples with responses generated by `Qwen/Qwen3-4B`.
> 
> ```yaml
> dataset_path: openai/gsm8k
> split: train
> model_path: Qwen/Qwen3-4B
> max_new_tokens: 1024
> temperature: 0.7
> top_p: 0.8
> top_k: 20
> ```
> 
> The prompt used to generate the teacher data was the same solver-style GSM8K prompt:
> 
> ```text
> You are a math solver. Given a math problem, your task is to compute the solution and return the final answer. Be concise and clear in your response.
> 
> {question}
> ```
> 
> ## 2. GSM8K-specific C2C Fuser Training
> 
> The receiver and sharer LMs were frozen; only the C2C projector parameters were trained.
> 
> ```yaml
> model:
>   base_model: Qwen/Qwen3-0.6B
>   teacher_model: Qwen/Qwen3-4B
>   include_response: false
>   is_do_alignment: false
>   alignment_strategy: first
>   mapping: last_aligned
>   projector:
>     type: C2CProjector
>     params:
>       hidden_dim: 1024
>       intermediate_dim: 1024
>       num_layers: 3
>       dropout: 0.1
>       initial_temperature: 1.0
>       final_temperature: 0.001
>       anneal_steps: 116
> 
> training:
>   learning_rate: 0.0001
>   weight_decay: 0.01
>   num_epochs: 1
>   max_length: 1024
>   gradient_accumulation_steps: 1
>   per_device_train_batch_size: 8
>   num_processes: 8
>   freeze: [teacher, base]
>   seed: 42
> 
> data:
>   type: LLMGeneratedChatDataset
>   train_ratio: 0.99
>   kwargs:
>     split: train
>     max_word_count: 1024
>     num_samples: null
>     data_path: <qwen3_4b_generated_gsm8k_train_dataset>
> ```
> 
> The training command was the standard C2C fuser training command:
> 
> ```bash
> torchrun --nproc_per_node=8 script/train/SFT_train.py \
>   --config <gsm8k_specific_fuser_training_config>
> ```
> 
> ## 3. Evaluation Setting for the 62.55 C2C Row
> 
> The evaluation used the full GSM8K test split and the GSM8K-specific fuser checkpoint from the training above.
> 
> ```yaml
> eval:
>   dataset: gsm8k
>   answer_method: generate
>   sample_interval: 1
>   use_template: true
> 
> model:
>   model_name: Rosetta
>   generation_config:
>     do_sample: false
>     max_new_tokens: 1024
>   rosetta_config:
>     base_model: Qwen/Qwen3-0.6B
>     teacher_model: Qwen/Qwen3-4B
>     checkpoints_dir: <gsm8k_specific_qwen3_0.6b_qwen3_4b_fuser>/final
>     include_response: false
>     is_do_alignment: false
>     alignment_strategy: first
> 
> chat_template:
>   add_generation_prompt: true
>   enable_thinking: false
> 
> prompt:
>   receiver_user_prompt: |
>     You are a math solver. Given a math problem, your task is to compute the solution and return the final answer. Be concise and clear in your response.
> 
>     {question}
>   generated_interpreter_text: none
> ```
> 
> With `dataset: gsm8k`, the evaluator uses the built-in GSM8K configuration:
> 
> - dataset: `openai/gsm8k`
> - split: `test`
> - sample size: full test split, 1319 examples when `sample_interval: 1` and no `limit` is set
> - metric: rule-based GSM8K final-answer equivalence using `GSM8KEvaluator.rule_judge`
> 
> The ground-truth answer is parsed from the GSM8K `answer` field after the `####` marker, and the generated answer is extracted and verified by the repository's math evaluator.
> 
> The evaluation command was:
> 
> ```bash
> python script/evaluation/unified_evaluator.py \
>   --config <gsm8k_c2c_eval_config>
> ```
> 
> Sorry again for the confusion in my earlier reply. The important correction is that the Table 15 62.55 C2C number depends on the GSM8K-specific fuser checkpoint.
>  

 _Originally posted by @minzh23 in [#13](https://github.com/thu-nics/C2C/issues/13#issuecomment-4707879280)_

Thanks for sharing the corrected GSM8K setup.

Would it be possible to update the C2C_Fuser repository and upload the GSM8K-specific Qwen3-4B → Qwen3-0.6B fuser checkpoint used for the Table 15 result? This checkpoint appears to be the most important missing artifact for reproducing the reported 62.55%.

I reproduced the pipeline from scratch and followed all disclosed teacher-data generation settings:

openai/gsm8k train split, all 7,473 examples

teacher: Qwen/Qwen3-4B

exact solver prompt from your comment

max_new_tokens: 1024

temperature: 0.7

top_p: 0.8

top_k: 20

I then trained a new task-specific fuser using the disclosed projector architecture, mapping, learning rate, weight decay, one-epoch schedule, seed 42, and 116-step temperature annealing.

On the full 1,319-example GSM8K test split, the result was:

451/1,319 = 34.19%

reported result: 825/1,319 = 62.55%

The only one remaining execution difference is that I used one A100 with batch size 8 and gradient accumulation 8, preserving the nominal effective batch size of 64 and 116 optimizer updates. The reported run used eight processes with batch size 8 and accumulation 1.

If possible, could you please upload:

the exact GSM8K-specific fuser checkpoint used for Table 15; and

optionally, the generated teacher dataset or exact teacher-generation command?

The checkpoint alone would already allow us to separate fuser-training differences from evaluation differences.

**Comment by minzh23** [comment]
Thanks for the detailed reproduction attempt! @zhangyun04 

We have now open-sourced the exact GSM8K-specific Qwen3-4B → Qwen3-0.6B C2C fuser checkpoint used for the 62.55% result in Table 15:

https://huggingface.co/nics-efc/C2C_Fuser/tree/main/qwen3_0.6b%2Bqwen3_4b_gsm8k_Fuser

This is the task-specific checkpoint used for the reported result, rather than the public general-purpose `qwen3_0.6b+qwen3_4b_Fuser` checkpoint.

We hope this checkpoint helps isolate fuser-training differences from evaluation differences. Thanks again for your careful reproduction work.
